src/Roave/SecurityAdvisories/FriendsOfPhp.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Roave.SecurityAdvisories.FriendsOfPhp (
+  parseAdvisoryYaml,
+  Advisory(..),
+  AffectedBranch(..)
+) where
+
+import Data.Map (Map)
+import Data.ByteString (ByteString)
+import Data.Text (Text)
+import Data.Yaml
+import qualified Data.Yaml as Y
+
+data AffectedBranch = AffectedBranch {
+  time :: Maybe Text
+  , versions :: [Text]
+} deriving (Eq, Show)
+
+data Advisory = Advisory {
+  title :: Maybe Text
+  , link :: Maybe Text
+  , cve :: Maybe Text
+  , branches :: Map Text AffectedBranch
+  , reference :: Maybe Text
+} deriving (Eq, Show)
+
+parseAdvisoryYaml :: ByteString -> Either String Advisory
+parseAdvisoryYaml = Y.decodeEither
+
+instance FromJSON AffectedBranch where
+  parseJSON (Y.Object v) =
+    AffectedBranch <$>
+    v .: "time" <*>
+    v .: "versions"
+  parseJSON _ = fail "Expected an Object to be parsed into an AffectedBranch"
```

Comment:
Not a fan of this kind of simplistic failure, but it works for now

Reply:
Generally not a `String`, but it is indeed the exception case (which I need to test)